packages/form/src/components/form/index.js

```diff
@@ -20,11 +20,14 @@ const Form = ( { children, name, onSubmit, ...props } ) => {
 		[ name ]
 	);
 
-	const { startSubmit, stopSubmit } = useDispatch( STORE_NAME );
+	const { startSubmit, stopSubmit, setFieldError } = useDispatch(
+		STORE_NAME
+	);
 
-	const registerValidation = ( fieldName, validation ) =>
-		( validations[ fieldName ] = validation );
+	const registerValidation = ( fieldClientId, validation ) =>
```

Comment:
This library is meant to be context-agnostic and is not only meant to be used by blocks.  
Like, that's the case right now but we'd also be using this for more complex setting/sharing/etc. forms throughout the dashboard as we migrate those views here.

Hence `fieldName` and not `fieldClientId` - nothing stops you from using a question's field `clientId` as the value though.

It's not even clear to me what the difference is between `fieldClientId` and using `fieldName` and just maybe updating the value to make it easier to work with?